### PR TITLE
support go build on  "GOARCH=wasm GOOS=js"

### PR DESCRIPTION
**Pull request by zhiquan911**
support `go build` on  "GOARCH=wasm GOOS=js"

### CODE REVIEW

**Comment by xiang90** [comment]
@zhiquan911 Can you resolve the merge conflict?

File: bolt_js.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+// mmap memory maps a DB's data file.
+func mmap(db *DB, sz int) error {
+	return nil
+}
+
+// munmap unmaps a DB's data file from memory.
+func munmap(db *DB) error {
+	return nil
+}
```
Comment (ahrtr) [inline bolt_js.go:35]: You do you exclude `js` in bolt_unix.go, so I guess you will run into compile errors?

**Comment by ahrtr** [comment]
Have you verified whether it work on `GOARCH=wasm GOOS=js` ? Please yes, could you show me the demo (e.g. sample program & output)?

**Comment by ahrtr** [comment]
Since there is no response from the original author anymore, so closing this PR. 

Please feel free to reopen it if anyone is interested in working on it.

**Pull request closed without merging**